pkg/gds/service_test.go

```diff
@@ -345,11 +344,26 @@ func (s *gdsTestSuite) loadFixtures(ftype fixtureType, fpath string) {
 		}
 	}
 
+	// Shutdown old GDS server
+	if s.svc != nil {
+		if err := s.svc.GetGDS().Shutdown(); err != nil {
+			log.Warn().Err(err).Msg("could not shutdown GDS server to start new one")
+		}
+	}
+	if s.grpc != nil {
+		s.grpc.Release()
+	}
+
 	// Create the new service with a database to the specified path
 	conf := gds.MockConfig()
 	conf.Database.URL = "leveldb:///" + s.dbPaths[ftype]
 	s.svc, err = gds.NewMock(conf)
 	require.NoError(err, "could not create mock GDS service")
+
+	// Start the new GDS server using a bufconn listener to avoid network requests
+	s.grpc = bufconn.New(bufSize)
+	go s.svc.GetGDS().Run(s.grpc.Listener)
+
```

Comment:
I was completed baffled for about 2 hours as to why the GDS database appeared to be closed when calling the `Register` RPC. Then I realized that we were replacing `s.svc` here with an entirely new service, which includes a brand new GDS, but the gRPC server was still pointing to the old GDS with the old database! :facepalm: So I've moved the code which starts the new gRPC server here instead of in `SetupTestSuite`.

Reply:
Ooft, that was definitely a tricky one. Great catch!